bindings/matrix-sdk-ffi/src/client.rs

```diff
@@ -1101,13 +1102,38 @@ fn gen_transaction_id() -> String {
 /// is dropped, the file will be removed from the disk.
 #[derive(uniffi::Object)]
 pub struct MediaFileHandle {
-    inner: SdkMediaFileHandle,
+    inner: RwLock<Option<SdkMediaFileHandle>>,
+}
+
+impl MediaFileHandle {
+    fn new(handle: SdkMediaFileHandle) -> Self {
+        Self { inner: RwLock::new(Some(handle)) }
+    }
 }
 
 #[uniffi::export]
 impl MediaFileHandle {
     /// Get the media file's path.
     pub fn path(&self) -> String {
-        self.inner.path().to_str().unwrap().to_owned()
+        self.inner
+            .read()
+            .unwrap()
+            .as_ref()
+            .expect("MediaFileHandle must not be used after calling persist")
+            .path()
+            .to_str()
+            .unwrap()
+            .to_owned()
+    }
+
+    pub fn persist(&self, path: String) -> bool {
+        let mut guard = self.inner.write().unwrap();
+        match guard.take().expect("MediaFileHandle was already persisted").persist(path.as_ref()) {
+            Ok(_) => true,
+            Err(e) => {
+                *guard = Some(e.file);
```

Comment:
What's this line for?

Reply:
We stole the handle a few lines above (with the `take` call), and in case of error, we put it back here. So if `persist` failed because of some temporary hiccup of the filesystem, for instance, we could try again later.